src/ui/components/Header/PlayingBarInfos.react.js

```diff
@@ -0,0 +1,172 @@
+import Player from '../../lib/player';
+import utils from '../../utils/utils';
+
+
+class PlayingBarInfos extends React.Component {
+  static propTypes = {
+    trackPlaying: PropTypes.object,
+    shuffle: PropTypes.bool,
+    repeat: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      elapsed: 0,
+      showTooltip: false,
+      duration: null,
+      x: null,
+      dragging: false,
+    };
+
+    this.tick = this.tick.bind(this);
+
+    this.dragOver = this.dragOver.bind(this);
+    this.dragEnd = this.dragEnd.bind(this);
+
+    this.jumpAudioTo = this.jumpAudioTo.bind(this);
+    this.showTooltip = this.showTooltip.bind(this);
+    this.hideTooltip = this.hideTooltip.bind(this);
+  }
+
+  componentDidMount() {
+    Player.getAudio().addEventListener('timeupdate', this.tick);
+
+    window.addEventListener('mousemove', this.dragOver);
+    window.addEventListener('mouseup', this.dragEnd);
+  }
+
+  componentWillUnmount() {
+    Player.getAudio().removeEventListener('timeupdate', this.tick);
+
+    window.removeEventListener('mousemove', this.dragOver);
+    window.removeEventListener('mouseup', this.dragEnd);
+  }
+
+  tick() {
+    this.setState({ elapsed: Player.getCurrentTime() });
+  }
+
+  jumpAudioTo(e) {
+    this.setState({ dragging: true });
+
+    const { trackPlaying } = this.props;
+
+    const bar = document.querySelector('.now-playing-bar');
+    const percent = ((e.pageX - (bar.offsetLeft + bar.offsetParent.offsetLeft)) / bar.offsetWidth) * 100;
+
+    const jumpTo = (percent * trackPlaying.duration) / 100;
+
+    AppActions.player.jumpTo(jumpTo);
+  }
+
+  dragOver(e) {
+    // Chack if it's needed to update currentTime
+    if (this.state.dragging) {
+      const { trackPlaying } = this.props;
+
+      const playingBar = this.refs.playingBar;
+      const playingBarRect = playingBar.getBoundingClientRect();
+
+      const barWidth = playingBar.offsetWidth;
+      const offsetX = Math.min(Math.max(0, e.pageX - playingBarRect.left), barWidth);
+
+      const percent = offsetX / barWidth * 100;
+
+      const jumpTo = (percent * trackPlaying.duration) / 100;
+
+      AppActions.player.jumpTo(jumpTo);
+    }
+  }
+
+  dragEnd() {
+    if (this.state.dragging) {
+      this.setState({ dragging: false });
+    }
+  }
+
+  showTooltip(e) {
+    const { trackPlaying } = this.props;
+
+    const offsetX = e.nativeEvent.offsetX;
+    const barWidth = e.currentTarget.offsetWidth;
+
+    const percent = offsetX / barWidth * 100;
+
+    const time = (percent * trackPlaying.duration) / 100;
+
+    this.setState({
+      duration: time,
+      x: percent,
+    });
+  }
+
+  hideTooltip() {
+    this.setState({
+      duration: null,
+      x: null,
+    });
+  }
+
+  render() {
+    const { trackPlaying } = this.props;
+    let elapsedPercent;
+
+    if (this.state.elapsed < trackPlaying.duration) elapsedPercent = this.state.elapsed * 100 / trackPlaying.duration;
```

Comment:
Why do we only compute that conditionally?

Reply:
This is a really good question, this looks weird